tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -73,5 +81,68 @@ internal static string PropagateSpanData(DbmPropagationLevel propagationStyle, s
 
             return string.Empty;
         }
+
+        /// <summary>
+        /// Uses a sql instruction to set a context for the current connection, bearing the span ID and trace ID.
+        /// This is meant to circumvent cache invalidation issues that occur when those values are injected in comment.
+        /// Currently only working for MSSQL (uses an instruction that is specific to it)
+        /// </summary>
+        /// <returns>True if the traceparent information was set</returns>
+        internal static bool PropagateDataViaContext(DbmPropagationLevel propagationLevel, IntegrationId integrationId, IDbConnection? connection, Span span)
+        {
+            if (propagationLevel != DbmPropagationLevel.Full || integrationId != IntegrationId.SqlClient || connection == null)
+            {
+                return false;
+            }
+
+            var stopwatch = System.Diagnostics.Stopwatch.StartNew();
+
+            const byte version = 0; // version can have a maximum value of 15 in the current format
+            var sampled = SamplingPriorityValues.IsKeep(span.Context.TraceContext.GetOrMakeSamplingDecision());
+            var contextValue = BuildContextValue(version, sampled, span.SpanId, span.TraceId128);
+
+            using (var injectionCommand = connection.CreateCommand())
+            {
+                injectionCommand.CommandText = SetContextCommand;
+
+                var parameter = injectionCommand.CreateParameter();
+                parameter.ParameterName = ContextInfoParameterName;
+                parameter.Value = contextValue;
+                parameter.DbType = DbType.Binary;
+                injectionCommand.Parameters.Add(parameter);
+
+                injectionCommand.ExecuteNonQuery();
+
+                if (Log.IsEnabled(LogEventLevel.Debug))
+                {
+                    // avoid building the string representation in the general case where debug is disabled
+                    Log.Debug("Span data for DBM propagated for {Integration} via context_info with value {ContextValue} (propagation level: {PropagationLevel}", integrationId, HexConverter.ToString(contextValue), propagationLevel);
+                }
+            }
+
+            // Since sending the query to the DB can be a bit long, we register the time it took for transparency.
+            // Not using _dd because we want the customers to be able to see that tag.
+            span.SetMetric("dd.instrumentation.time_ms", stopwatch.Elapsed.TotalMilliseconds);
```

Comment:
Is `dd.instrumentation.time_ms` just a one-off name or is this something common? never seen it before and search doesn't show any other usages anywhere in the org

Reply:
hmm yeah it's a one off, I'm actually not aware of any existing instrumentation that would be long enough that we'd want to document how long it takes.